src/foam/u2/view/RichChoiceView.js

```diff
@@ -725,9 +725,8 @@ foam.CLASS({
             'text-overflow': 'ellipsis'
           });
 
-          return this.add(this.fullObject$.map(o => {
-            return ( o && o.toSummary() ) || this.defaultSelectionPrompt;
-          }));
+          return this.fullObject && this.fullObject.toSummary() ?
```

Comment:
Should store the result of this.fullObject.toSummary() in a var so you don't have to calculate it three times.